source/routing/tests/circle_test.cpp

```diff
@@ -0,0 +1,103 @@
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+Stay tuned using
+twitter @navitia
+IRC #navitia on freenode
+https://groups.google.com/d/forum/navitia
+www.navitia.io
+*/
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE circle_test
+
+#include "type/geographical_coord.h"
+#include "routing/circle.h"
+#include <boost/test/unit_test.hpp>
+#include <iomanip>
+#include <vector>
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/polygon.hpp>
+#include <stdlib.h>
+#include <time.h>
+
+
+using namespace navitia::routing;
+
+BOOST_AUTO_TEST_CASE(project_in_direction_test) {
+    navitia::type::GeographicalCoord coord_Paris = navitia::type::GeographicalCoord(2.3522219000000177,48.856614);
+    navitia::type::GeographicalCoord coord_North = navitia::type::GeographicalCoord(90,0);
+    navitia::type::GeographicalCoord coord_Equator = navitia::type::GeographicalCoord(0,48.856614);
+    BOOST_CHECK_CLOSE (project_in_direction(coord_Paris, 80, 10).distance_to(coord_Paris), 10, 0.1);
+    BOOST_CHECK_CLOSE (project_in_direction(coord_Paris, 115, 42).distance_to(coord_Paris), 42, 0.1);
+    BOOST_CHECK_CLOSE (project_in_direction(coord_Paris, 195, 115).distance_to(coord_Paris), 115, 0.1);
+    BOOST_CHECK_CLOSE (project_in_direction(coord_Paris, 314, 29).distance_to(coord_Paris), 29, 0.5);
+    // Check for the angle with exceptions
+    BOOST_CHECK_CLOSE (project_in_direction(coord_Paris, 90, 42).distance_to(coord_Paris), 42, 0.1);
+    BOOST_CHECK_CLOSE (project_in_direction(coord_Paris, 180, 12).distance_to(coord_Paris), 12, 0.5);
+    BOOST_CHECK_CLOSE (project_in_direction(coord_Paris, 270, 89).distance_to(coord_Paris), 89, 0.1);
+    BOOST_CHECK_CLOSE (project_in_direction(coord_Paris, 360, 1415).distance_to(coord_Paris), 1415, 0.1);
+    // Check when the direction is not in [0,360]
+    BOOST_CHECK_CLOSE (project_in_direction(coord_Paris, 400, 568).distance_to(coord_Paris), 568, 0.5);
+    BOOST_CHECK_CLOSE (project_in_direction(coord_Paris, -36,1000).distance_to(coord_Paris), 1000, 0.5);
+    BOOST_CHECK_CLOSE (project_in_direction(coord_Paris, -667, 87).distance_to(coord_Paris), 87, 0.5);
+    //Check on the pole and on the equator
+    BOOST_CHECK_CLOSE (project_in_direction(coord_North, 30, 2016).distance_to(coord_North), 2016, 0.5);
+    BOOST_CHECK_CLOSE (project_in_direction(coord_Equator, 70, 1993).distance_to(coord_Equator), 1993, 0.5);
+}
+
+BOOST_AUTO_TEST_CASE(circle_test) {
+    navitia::type::GeographicalCoord coord_Paris = navitia::type::GeographicalCoord(2.3522219000000177, 48.856614);
+    navitia::type::GeographicalCoord coord_Pekin = navitia::type::GeographicalCoord(-89.61, 40.5545);
+    navitia::type::GeographicalCoord coord_Almost_North = navitia::type::GeographicalCoord(180, 0.5);
+    navitia::type::GeographicalCoord coord_North = navitia::type::GeographicalCoord(268, 0);
+    navitia::type::Polygon c_Paris_42=circle(coord_Paris, 42);
```

Comment:
you need to put spaces around the `=`

and it's up to you, but you can replace `navitia::type::Polygon` with `auto`